sql/item_strfunc.cc

```diff
@@ -1099,9 +1099,9 @@ String *Item_func_des_decrypt::val_str(String *str)
     THD *thd= current_thd;
     push_warning_printf(thd, Sql_condition::WARN_LEVEL_WARN,
                         ER_FEATURE_DISABLED, ER_THD(thd, ER_FEATURE_DISABLED),
-                        "des_decrypt", "--with-ssl");
+                        "des_decrypt", "-DHAVE_des");
```

Comment:
This is only marginally better. One cannot run `cmake -DHAVE_des`, it'll be an error. One can run `cmake -DHAVE_des=1` but it will fail to compile with OpenSSL doesn't have DES.

Which is why I suggested the variant that mentions "OpenSSL compiled with DES support"

Reply:
the point is that even if one has openssl with DES support, mariadb may have been compiled with `-DHAVE_des=OFF` to turn that off. And I don't know if it was intentionally turned off, or openssl doesn't have it.

Also I don't want to break translations and have custom string there, as current string has this placeholders

```
ER_FEATURE_DISABLED 
        eng "The '%s' feature is disabled; you need MariaDB built with '%s' to have it working"
```

Reply:
Also, what does `--with-ssl` refer to? as that's not a valid key for cmake is it?

Reply:
I do prefer people trying to turn on HAVE_des and finding out that their openssl doesn't have, and that it doesn't get detected as on automatically. As that's something one can grep for, and find the right code, commit, cmake options.

Reply:
Okay, as you like. My reviewer's approval still holds.

`--with-ssl` refers to `./configure --with-ssl` from the times when MySQL was using autotools and it was still possible to compile MySQL without SSL support. Both are long in the past.

Reply:
Really do not want to break translations. So how about this:

> The 'des_encrypt' feature is disabled; you need MariaDB built with 'openssl des cipher (HAVE_des)' to have it working

Reply:
better, thanks